app-modules/prospect/tests/Prospect/EditProspectTest.php

```diff
@@ -149,3 +152,92 @@
             data: ['recordId' => $basicNeedsProgram->getKey()]
         )->assertSuccessful();
 });
+
+test('convert action visible when prospect is not converted to student', function () {
+    $user = User::factory()->licensed(Prospect::getLicenseType())->create();
+
+    $prospect = Prospect::factory()->create();
+
+    $user->givePermissionTo('prospect.view-any');
+    $user->givePermissionTo('prospect.*.update');
+
+    actingAs($user);
+
+    livewire(EditProspect::class, [
+        'record' => $prospect->getRouteKey(),
+    ])
+        ->assertSuccessful()
+        ->assertActionVisible(ConvertToStudent::class)
+        ->assertActionHidden(DisassociateStudent::class);
+});
+
+test('disassociate student action visible when prospect is converted to student', function () {
+    $user = User::factory()->licensed([Prospect::getLicenseType(), Student::getLicenseType()])->create();
+
+    $prospect = Prospect::factory()
+        ->for(Student::factory(), 'student')
+        ->create();
+
+    $user->givePermissionTo('prospect.view-any');
+    $user->givePermissionTo('prospect.*.update');
+
+    actingAs($user);
+
+    livewire(EditProspect::class, [
+        'record' => $prospect->getRouteKey(),
+    ])
+        ->assertSuccessful()
+        ->assertActionVisible(DisassociateStudent::class)
+        ->assertActionHidden(ConvertToStudent::class);
+});
+
+test('convert prospect to student', function () {
+    $user = User::factory()->licensed([Prospect::getLicenseType(), Student::getLicenseType()])->create();
+
+    $user->givePermissionTo('prospect.view-any');
+    $user->givePermissionTo('prospect.*.update');
+
+    actingAs($user);
+
+    $prospect = Prospect::factory()
+        ->create();
+
+    $student = Student::factory()
+        ->create();
+
+    livewire(EditProspect::class, [
+        'record' => $prospect->getRouteKey(),
+    ])
+        ->mountAction(ConvertToStudent::class)
+        ->callAction(
+            ConvertToStudent::class,
+            data: ['student_id' => $student->getKey()]
+        )->assertSuccessful();
+
+    $prospect->refresh();
+    expect($prospect->student()->exists())->toBeTrue();
```

Comment:
You should assert that the relationship is to the `Student` that you expect it to be